src/buttons/close.ts

```diff
@@ -0,0 +1,38 @@
+import { ButtonBuilder, ButtonStyle } from "discord.js";
+import InteractionContext from "../models/InteractionContext";
+import NeedleButton from "../models/NeedleButton";
+import NeedleBot from "../NeedleBot";
+import ObjectFactory from "../ObjectFactory";
+import CommandExecutorService from "../services/CommandExecutorService";
+
+export default class CloseButton extends NeedleButton {
+	public readonly customId = "close";
+	private readonly commandExecutor: CommandExecutorService;
+
+	constructor(bot: NeedleBot) {
+		super(bot);
+		this.commandExecutor = ObjectFactory.createCommandExecutorService();
+	}
+
+	public getBuilder(): ButtonBuilder {
+		return new ButtonBuilder()
+			.setCustomId(this.customId)
+			.setLabel("Archive thread")
+			.setStyle(ButtonStyle.Success)
+			.setEmoji("1010182198923636797"); // :archive_3_0:
+	}
+
+	public async press(context: InteractionContext): Promise<void> {
+		if (!context.isInGuild()) return;
+
+		const closeCommand = this.bot.getCommand(this.customId);
+		const { interaction, settings, replyInSecret } = context;
+		const { member, channel } = interaction;
+		const hasPermission = await closeCommand.hasPermissionToExecuteHere(member, channel);
+		if (!hasPermission) {
+			return replyInSecret(settings.ErrorInsufficientUserPerms);
```

Comment:
I don't think I've ever used the debugger in JS/TS 🙊 Can revisit later but it would be a change in many places, not going to happen in this PR

Reply:
Excellent followup then.

Reply:
Yea I know, but I really have not noticed a big difference when developing (and it's kind of mixed where we do this and where we don't right now). It's too much of a headache to figure out how we should do this across the entire code base now, as it will also change how we handle errors (either try/catch or Promise.catch which will be affected when we await or not await things). Too risky for this PR but I'm happy to do it in the future in some refactoring PR